internal/schedule_client.go

```diff
@@ -476,6 +476,10 @@ type (
 	ScheduleUpdate struct {
 		// Schedule - New schedule to replace the existing schedule with
 		Schedule *Schedule
+
+		// SearchAttributes - Optional indexed info that can be used for querying via the List schedules APIs.
+		// The key and value type must be registered on Temporal server side.
+		SearchAttributes *SearchAttributes
```

Comment:
Don't think we need to take a pointer here. Also what is the behaviour if `SearchAttributes ` is null? I assume it leaves the `SearchAttributes` unchanged?

Reply:
Maybe I misunderstood, but I seem to recall the pointer being a suggestion from @cretz . @cretz -- am I off base here?

Yes, null means unchanged.

Reply:
Pointers default to nil, which can be useful to indicate that `SearchAttributes` has not been set, as opposed to having an empty but allocated struct. The behavior I would expect:

nil: leave current SAs untouched
empty: remove current SAs

I don't know if there is precedence for this pattern, though.

Reply:
A part of me would expect this to operate as an idempotent HTTP PUT REST(ish) resource, which effectively replaces the resource and has a symmetrical contract with GET (i.e. describe).

Reply:
Sadly, it is the way it is because SearchAttributes was a later addition to ScheduleUpdate (and Memo will be another addition). The Schedule field works as you think it should, but we couldn't make SearchAttributes work that way too since existing clients aren't sending it.

And yes, there should be some way (at the Go level) to unset all custom search attributes. And empty (but non-nil) map seems like the most straightforward way, but there could be others.

Reply:
Hrmm, I don't recall making this suggestion, but it's possible I did. And looking at it, it makes sense. Unfortunately for this call, we need to differentiate between unset (i.e. don't change) and empty (i.e. actually empty).

Reply:
Thanks. I'll go ahead and add tests to ensure:
1. A nil SearchAttributes leaves SAs untouched.
2. An empty SearchAttributes removes any pre-existing SAs.

Reply:
Added tests